openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -142,12 +155,25 @@ public void createComponentSchema(Map<String, Schema> schema, TypeSymbol typeSym
                 break;
             case MAP:
                 MapTypeSymbol mapTypeSymbol = (MapTypeSymbol) type;
-                TypeDescKind typeDescKind = mapTypeSymbol.typeParam().typeKind();
-                Schema openApiSchema = ConverterCommonUtils.getOpenApiSchema(typeDescKind.getName());
-                schema.put(componentName,
-                        new ObjectSchema().additionalProperties(
-                                openApiSchema.getType() == null ? true : openApiSchema)
-                                .description(typeDoc));
+                TypeSymbol typeParam = mapTypeSymbol.typeParam();
+
+                if (typeParam.typeKind() == TypeDescKind.TYPE_REFERENCE) {
```

Comment:
Will this break for the non-simple reference map type? (e.g.: inline return as the map).

Reply:
Do we mean something like below ?
```bal
import ballerina/http;

service /payloadV on new http:Listener(9090) {

    resource function get res3(string id) returns map<string> {
        return {
            "id": "001"
        };
    }
}
```

In this kind of cases the code flow is not breaking. 

Generated response

```yaml
responses:
        "200":
          description: Ok
          content:
            application/json:
              schema:
                type: object
                additionalProperties:
                  type: string
```